src/Microsoft.Identity.Web/MicrosoftGraph/MicrosoftGraphServiceExtensions.cs

```diff
@@ -0,0 +1,93 @@
+            this MicrosoftIdentityAppCallsWebApiAuthenticationBuilder builder,
+            IConfigurationSection configurationSection)
+        {
+            return builder.AddMicrosoftGraphServiceClient(
+                options => configurationSection.Bind(options));
+        }
+
+        /// <summary>
+        /// Add support to calls Microsoft graph. From a base graph Url and a default scope.
+        /// </summary>
+        /// <param name="builder">Builder.</param>
+        /// <param name="graphBaseUrl">Named instance of option.</param>
+        /// <param name="defaultScopes">Configuraiton section.</param>
+        /// <returns>The builder to chain.</returns>
+        public static MicrosoftIdentityAppCallsWebApiAuthenticationBuilder AddMicrosoftGraphServiceClient(
+            this MicrosoftIdentityAppCallsWebApiAuthenticationBuilder builder,
+            string graphBaseUrl = "https://graph.microsoft.com/v1.0",
+            string defaultScopes = "user.read")
+        {
+            return builder.AddMicrosoftGraphServiceClient(
+                options =>
+                {
+                    options.BaseUrl = graphBaseUrl;
+                    options.Scopes = defaultScopes;
+                });
+        }
+
+        /// <summary>
+        /// Add support to calls Microsoft graph. From a named options and a configuraiton method.
+        /// </summary>
+        /// <param name="builder">Builder.</param>
+        /// <param name="configureMicrosoftGraphOptions">Method to configure the options.</param>
+        /// <returns>The builder to chain.</returns>
+        public static MicrosoftIdentityAppCallsWebApiAuthenticationBuilder AddMicrosoftGraphServiceClient(
+            this MicrosoftIdentityAppCallsWebApiAuthenticationBuilder builder,
+            Action<MicrosoftGraphOptions> configureMicrosoftGraphOptions)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            // https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
+            builder.Services.AddOptions<MicrosoftGraphOptions>().Configure(configureMicrosoftGraphOptions);
+            builder.Services.AddTokenAcquisition(true);
+
+            builder.Services.AddSingleton<GraphServiceClient, GraphServiceClient>(serviceProvider =>
+            {
+                var tokenAquisitionService = serviceProvider.GetRequiredService<ITokenAcquisition>();
+                var options = serviceProvider.GetRequiredService<IOptions<MicrosoftGraphOptions>>();
+
+                var microsoftGraphOptions = options.Value;
+                if (microsoftGraphOptions.Scopes == null)
+                {
+                    throw new ArgumentException("CalledApiScopes should not be null.");
+                }
```

Comment:
this needs to move to the const file section for exceptions

Reply:
fixed in this [PR](https://github.com/AzureAD/microsoft-identity-web/pull/442)

---
In reply to: [469488173](https://github.com/AzureAD/microsoft-identity-web/pull/434#discussion_r469488173) [](ancestors = 469488173)